### PR TITLE
fix: port various changes from weasel environment repository

### LINKED ISSUE #4865: jx boot not creating valid tls certificates
### Summary 
Installing with jx boot and tls enabled results in wildcard certificates with an invalid issuer.

Inspecting the certificates shows an issuer of `cert-manager.local`

The cert-manager log contains many entries of the form:
```
E0726 11:43:44.078591       1 controller.go:200] cert-manager/controller/orders "msg"="re-queuing item  due to error processing" "error"="error creating new order: acme: urn:ietf:params:acme:error:malformed: Error creating new order :: Domain name \"hook-jx.kill-9.uk\" is redundant with a wildcard domain in the same request. Remove one or the other from the certificate request." "key"="jx/tls-kill-9-uk-p-1880648896"
```

### Jx version

The output of `jx version` is:

```
jx                 2.0.513
Kubernetes cluster v1.12.8-gke.10
kubectl            v1.14.2
helm client        Client: v2.14.0+g05811b8
git                git version 2.21.0
Operating System   Mac OS X 10.13.6 build 17G6030
```

### LINKED ISSUE #5190: Backport the fixes from environment-tekton-weasel-dev to jenkins-x-boot-config
### Summary 

Backport all fixes and changes from environment-tekton-weasel-dev to jenkins-x-boot-config.

### Actual behavior

The config seems to be out of sync, and the  environment-tekton-weasel-dev contains important fixes. 

### Jx version

The output of `jx version` is:

```
COPY OUTPUT HERE
```

### Jenkins type

- [ x] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### PR DESCRIPTION
**Pull request by ccojocar**
Only the relevant changes are moved over. These are changes which are not specific to
jenkins-x production environment (environment-tekton-weasel-dev).

fixes jenkins-x/jx#5190
fixes jenkins-x/jx#4865

### CODE REVIEW

File: env/templates/700-hook-ing.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ metadata:
   name: hook
   annotations:
     kubernetes.io/ingress.class: nginx
+{{- if eq .Values.certmanager.production "true" }}
+    certmanager.k8s.io/issuer: letsencrypt-prod
+{{- else }}
+    certmanager.k8s.io/issuer: letsencrypt-staging
+{{- end }}
```
Comment (cagiti) [inline env/templates/700-hook-ing.yaml]: This if/else/end block is no longer required within the ingress

File: env/parameters.schema.json [hunk 2]
```diff
@@ -40,11 +39,6 @@
           "title": "Pipeline bot Git username",
           "description": "The Git user that will perform git operations inside a pipeline. It should be a user within the Git organisation/owner where environment repositories will live. This is normally a bot."
         },
-        "email": {
```
Comment (cagiti) [inline env/parameters.schema.json]: Looks like some changes are being reverted, possibly just needs rebasing.

Comment (ccojocar) [reply env/parameters.schema.json]: I am not sure, but it seems to be removed also form here https://github.com/jenkins-x/environment-tekton-weasel-dev/blob/676ef0e92724ed89ae6193aa5a7ac20a2b01b1f8/env/parameters.schema.json#L31

Was this accidentally done?

File: jenkins-x.yml [hunk 3]
```diff
@@ -19,43 +19,51 @@ pipelineConfig:
...
                   - name: DEPLOY_NAMESPACE
                     value: kube-system
-              - name: helm-cluster-values
+              - name: create-install-values
                 dir: /workspace/source/env
                 command: jx
                 args: ['step','create','install', 'values', '-b']
-              - name: install-vault
-                dir: /workspace/source/systems/vault
+              - name: install-external-dns
```
Comment (ccojocar) [inline jenkins-x.yml:36]: @cagiti This step needs to be conditionally executed only when `externalDNS` is enabled. The BDD tests are currently failing when the externalDNS is disabled. I am looking to find a solution for this.

File: env/templates/700-hook-ing.yaml (hunk 1 above)
Comment (ccojocar) [reply env/templates/700-hook-ing.yaml]: It seems to be present on https://github.com/jenkins-x/environment-tekton-weasel-dev/blob/676ef0e92724ed89ae6193aa5a7ac20a2b01b1f8/env/templates/700-hook-ing.yaml#L8. 

Should be removed from all ingress resources? Can we use only `certmanager.k8s.io/issuer: letsencrypt-prod`?

Comment (cagiti) [reply env/templates/700-hook-ing.yaml]: Yes, should not be removed... I don't think the PR that James R raised on the boot config was merged before the weasel environment was created

File: env/parameters.schema.json (hunk 2 above)
Comment (cagiti) [reply env/parameters.schema.json]: I think the weasel environment was created before this PR was merged https://github.com/jenkins-x/jenkins-x-boot-config/pull/44/files

File: jenkins-x.yml (hunk 3 above)
Comment (cagiti) [reply jenkins-x.yml:36]: yeah, would be good to perhaps wrap the contents of the files that step executes with a condition so if you haven't got it enables it basically no-ops

File: env/templates/700-hook-ing.yaml (hunk 1 above)
Comment (rawlingsj) [reply env/templates/700-hook-ing.yaml]: yeah that block can be removed as we use external dns and the new issuer format https://github.com/jenkins-x/environment-tekton-weasel-dev/blob/676ef0e92724ed89ae6193aa5a7ac20a2b01b1f8/env/templates/cert-manager-prod-issuer.yaml#L13-L26 this means annotations on ingress rules aren't needed.

Comment (ccojocar) [reply env/templates/700-hook-ing.yaml]: Should I remove it on all ingress resources?

File: env/parameters.schema.json (hunk 2 above)
Comment (ccojocar) [reply env/parameters.schema.json]: I reverted the changes to include the `email` field.

File: jenkins-x.yml (hunk 3 above)
Comment (ccojocar) [reply jenkins-x.yml:36]: Any idea how is the best to do this. The values is defined in `jx-requirements.yml` and I don't see how the values are propagated into the `jenkins-x.yml` pipeline file. 

It seems that some commands are loading the `jx-requirements.yml` file and are parsing a specific key from it. e.g `step boot vault` is looking for `secretStore: vault`.

File: env/templates/700-hook-ing.yaml (hunk 1 above)
Comment (cagiti) [reply env/templates/700-hook-ing.yaml]: yes, if it's present 👍

Comment (ccojocar) [reply env/templates/700-hook-ing.yaml]: It was only on hook. All ingress should not have now any cert-manager annotations.

File: jenkins-x.yml [hunk 4]
```diff
@@ -19,43 +19,50 @@ pipelineConfig:
...
+                command: "if $(awk '/^.*externalDNS.*/ {print $2}' ../../jx-requirements.yml); then jx step helm apply --boot --remote --no-vault --name jx; else exit 0; fi"
+              - name: install-cert-manager-crds
                 dir: /workspace/source
                 command: kubectl
                 args: ['apply', '--wait', '--validate=true', '-f', 'https://raw.githubusercontent.com/jetstack/cert-manager/release-0.8/deploy/manifests/00-crds.yaml']
                 env:
                   - name: DEPLOY_NAMESPACE
                     value: cert-manager
-              - name: cert-manager
+              - name: install-cert-manager
```
Comment (ccojocar) [inline jenkins-x.yml:47]: also this step needs to be excluded when tls is disabled

**Comment by abayer** [comment]
The `jenkins-x.yml` was busted, but I've just now fixed it.

**Comment by abayer** [comment]
The failures now seem legit and aren't due to `jenkins-x.yml` format issues. =)

**Comment by ccojocar** [comment]
> The failures now seem legit and aren't due to jenkins-x.yml format issues. =)

yeah, that was mistakenly removed but still fails because of the empty charts (e.g. when cert-manager and external-dns are disabled). 

This fix is required in the builder-go https://github.com/jenkins-x/jx/pull/5223.

**Comment by ccojocar** [comment]
The `builder-go` still needs to be updated. I am waiting to be released with the latest fixes from jx.

**Pull request merged**